x-pack/legacy/plugins/reporting/server/routes/lib/authorized_user_pre_routing.ts

```diff
@@ -5,49 +5,37 @@
  */
 
 import Boom from 'boom';
-import { Legacy } from 'kibana';
+import { KibanaRequest } from 'src/core/server';
 import { AuthenticatedUser } from '../../../../../../plugins/security/server';
 import { ReportingConfig } from '../../../server';
 import { LevelLogger as Logger } from '../../../server/lib';
-import { ReportingSetupDeps } from '../../../server/types';
 import { getUserFactory } from '../../lib/get_user';
+import { ReportingSetupDeps } from '../../types';
 
 const superuserRole = 'superuser';
 
 export type PreRoutingFunction = (
-  request: Legacy.Request
+  request: KibanaRequest
 ) => Promise<Boom<null> | AuthenticatedUser | null>;
 
 export const authorizedUserPreRoutingFactory = function authorizedUserPreRoutingFn(
   config: ReportingConfig,
   plugins: ReportingSetupDeps,
   logger: Logger
 ) {
-  const getUser = getUserFactory(plugins.security, logger);
-  const { info: xpackInfo } = plugins.__LEGACY.plugins.xpack_main;
-
-  return async function authorizedUserPreRouting(request: Legacy.Request) {
-    if (!xpackInfo || !xpackInfo.isAvailable()) {
-      logger.warn('Unable to authorize user before xpack info is available.', [
-        'authorizedUserPreRouting',
-      ]);
-      return Boom.notFound();
-    }
-
-    const security = xpackInfo.feature('security');
-    if (!security.isEnabled() || !security.isAvailable()) {
-      return null;
-    }
+  const getUser = getUserFactory(plugins.security);
 
-    const user = await getUser(request);
+  return function authorizedUserPreRouting(request: KibanaRequest) {
+    const user = getUser(request);
 
     if (!user) {
-      return Boom.unauthorized(`Sorry, you aren't authenticated`);
+      throw Boom.unauthorized(`Sorry, you aren't authenticated`);
```

Comment:
Ideally the returned `authorizedUserPreRouting ` would have the `KibanaResponseFactory` as a parameter to avoid throwing boom error and using `res.xxxError` instead.